### DISCUSSION TITLE
Editing an input in a table

### DISCUSSION THREAD

**Discussion by lgalloherran** [twilio-labs/paste#1532] [Office hours] [answered]
Hey Dsys Team! 
i'm working on updating some legacy pages to Paste and I found that the 'edit' option for one input is no longer available in Paste so I wanted to ask you what are the alternatives to this, besides a modal. 
Here is the prototype that has the same functionality as the Legacy version for you to get an idea of how it is today: 

https://www.figma.com/proto/xXHiWJjfaifvNECs9B4yt2/Verifier-Caller-IDs-Paste-mocks?page-id=0%3A1&node-id=4%3A8207&viewport=2122%2C2892%2C0.6599365472793579&scaling=min-zoom

**You have to clic on the input 'friendly name'**

**Comment by serifluous** [comment]
Hey @lgalloherran, just to get some context on what alternatives we might suggest: Is there a customer problem you're trying to solve where the Modal wouldn't work, or are you looking to do some exploration into the interaction in general? (Both reasons are totally valid!)

**Reply by lgalloherran** [reply]
We want to explore other options cause we think that opening a modal for just one input is too much.

**Comment by richbachman** [comment, marked as answer]
Hi. Thanks for coming to office hours this week. Here's a link to the recording https://www.loom.com/share/bce26e8d326c4e07bf200d51d9009e4b, the passcode for which can be found in [Slack](https://twilio.slack.com/archives/CBEQHHDLP/p1623351563078300).

**Reply by lgalloherran** [reply]
Thanks! 
Just to follow-up on our discussion, I update the mocks with the pop-over option. 
I put the option with the edit button under 'actions' column, and another one with the edit icon under the friendly name column. 
Any thoughts? 

![Captura de Pantalla 2021-06-11 a la(s) 5 00 41 p  m](https://user-images.githubusercontent.com/72419222/121752889-c90fcc80-cad6-11eb-986f-dca4d343d416.png)
![Captura de Pantalla 2021-06-11 a la(s) 4 59 17 p  m](https://user-images.githubusercontent.com/72419222/121752892-cad99000-cad6-11eb-8757-466109a73c75.png)

**Reply by kate-porter** [reply]
Hey @lgalloherran - this treatment looks great! I'm having trouble spotting the difference between the two screenshots you shared though—I think you might have shared the same one twice (or I need some help spotting the difference 😂).

**Reply by lgalloherran** [reply]
Oh, I just realized that I repeated the same option! 
This is the other option ;P 
<img width="948" alt="Captura de Pantalla 2021-06-16 a la(s) 4 18 25 p  m" src="https://user-images.githubusercontent.com/72419222/122295154-7bbca200-cebe-11eb-875c-56c60d4c93fb.png">

**Reply by kate-porter** [reply]
Ah, got it, thank you!

I think I lean towards the option with the pencil icon beside the name, because it's super clear what I'll be editing when I click it. Whereas the "edit" button doesn't provide that clarity until after interacting with it.

I was just thinking, another nice approach that would scale well if you ever add more actions to this table would be to put an action menu (like [this one](https://paste.twilio.design/components/menu#choosing-a-menu-trigger)) in that "actions" column, and then you could have a "delete" action and an "edit friendly name" action in that menu. Clicking "edit friendly name" could then open a modal with an input to edit the name.

I think either approach works, but wanted to share that idea, too.

**Reply by lgalloherran** [reply]
Thanks for the suggestions. I think for now we are going with the pencil and the 'delete' option because we don't have any other actions that can be performed. But that's a good suggestions if once we will having more options :)
